lib/fs.js

```diff
@@ -15,6 +15,21 @@ const EventEmitter = require('events');
 const FSReqWrap = binding.FSReqWrap;
 const FSEvent = process.binding('fs_event_wrap').FSEvent;
 
+// TODO(ChALkeR,jasnell): remove the following try/catch in master after 7.x
+// Check for re-evaluation of the module. Doing so will cause the internal
+// module to fail to load. This is required to give users a grace period before
+// actually breaking modules that re-evaluate fs sources from context where
+// internal modules are not allowed, e.g. older versions of graceful-fs module.
```

Comment:
Note that this is still valid (even with the most recent graceful-fs@3 update) for users of graceful-fs@2, older versions of graceful-fs@3, and perhaps some user code that does the same thing (unlikely, but still).

Removing the warning would force us to introduce it again at least for a full major cycle in order to land other PRs that are blocked by re-evaluating `fs`.

So, please don't remove this on updates =).